doc/nrf/security/tfm.rst

```diff
@@ -78,131 +90,134 @@ A minimal version of the TF-M secure application is provided in |NCS| to show ho
-
-  * AES output feedback (AES-OFB) mode.
-  * AES cipher feedback (AES-CFB) mode.
-
-* Isolation level 3 is not supported.
-* In Isolation level 2 or higher, the number of peripherals configured as secure in Application Root of Trust (ARoT) is limited by the number of available MPU regions.
-* Nordic Semiconductor devices only support the GCC toolchain for building TF-M.
+.. _tfm_configurable_build:
+
+Configurable build
+==================
+
+To disable the minimal version of TF-M, set the :kconfig:option:`CONFIG_TFM_PROFILE_TYPE_NOT_SET` Kconfig option or one of the other build profiles.
+For description of the build profiles, see :ref:`tf-m_profiles`.
+It is not recommended to use predefined TF-M profiles as they might result in a larger memory footprint than necessary.
+
+When the :kconfig:option:`CONFIG_TFM_PROFILE_TYPE_NOT_SET` Kconfig option is enabled, the build process will not set a specific TF-M profile type.
+This allows for a more flexible configuration where individual TF-M features can be enabled or disabled as needed.
+It also provides more control over the build process and allows for a more fine-grained configuration of the TF-M secure image.
+
+To configure the features of the TF-M secure image, you must choose which TF-M partitions and which secure services to include in the build.
+A "TF-M partition" in this context refers to a secure partition within the Trusted Firmware-M architecture.
+These partitions are isolated from each other and from the non-secure application code.
+A service running inside TF-M would typically be implemented within one of these secure partitions.
+Each service can be a separate partition, or multiple related services might be grouped into a single partition.
+The partition provides the execution environment for the service, handling secure function calls and ensuring that the service's code and data are protected from unauthorized access.
+
+Following are the available Kconfig options for TF-M partitions:
+
+.. list-table:: Available TF-M Partitions
+   :header-rows: 1
+
+   * - Option Name
+     - Description
+     - Default Value
+     - Dependencies
+   * - :kconfig:option:`CONFIG_TFM_PARTITION_PLATFORM`
+     - Provides platform services.
+     - Enabled
+     -
+   * - :kconfig:option:`CONFIG_TFM_PARTITION_CRYPTO`
+     - Provides cryptographic services.
+     - Enabled
+     - INTERNAL_TRUSTED_STORAGE
+   * - :kconfig:option:`CONFIG_TFM_PARTITION_PROTECTED_STORAGE`
+     - Provides secure storage services.
+     - Enabled
+     - PLATFORM, CRYPTO
+   * - :kconfig:option:`CONFIG_TFM_PARTITION_INTERNAL_TRUSTED_STORAGE`
+     - Provides internal trusted storage services.
+     - Enabled
+     -
+   * - :kconfig:option:`CONFIG_TFM_PARTITION_INITIAL_ATTESTATION`
+     - Provides initial attestation services.
+     - Disabled
+     - CRYPTO
+
+Security Partition Manager (SPM) backend may also be configured, depending on the isolation requirements of the application.
```

Comment:
```suggestion
Security Partition Manager (SPM) backend may also be configured, depending on the isolation requirements of the application.

.. note::
    Do not confuse _Security_ Partition Manager with _Secure_ Partition Manager, which was removed in the |NCS| v2.1.0. See :ref:`migration guide <spm_to_tfm_migration>`.
```

Reply:
Sorry, but this is incorrect.

We have ended up into a conflicting acronyms. Secure Partition Manager as used by Nordic in previous SDK versions has nothing to do with [Secure Partition Manager](https://tf-m-user-guide.trustedfirmware.org/design_docs/services/secure_partition_manager.html) as defined by TF-M. Both are SPM, and they do not mean anything similar.